src/runtime_src/core/common/shim/hwctx_handle.h

```diff
@@ -53,6 +53,9 @@ class hwctx_handle
   virtual slot_id
   get_slotidx() const = 0;
 
+  virtual xrt::uuid
+  get_xclbin_uuid() const = 0;
```

Comment:
This will break all existing shims, and may in fact not be supported by all shims.   It would be better to default the implementation to throw an exception (not implemented or not supported).

Currently xrt::hw_context::get_xclbin_uuid() is implemented in the opaque impl class, not in shim.  I am not objecting to this change, just mentioning that it will not work without many changes.

I am also looking for where this function is even implemented in edge in this PR?

I want to mention that from a code architecture point of view `xrt_core -- depends on --> xrt_coreutil`, which implies that xrt_core can use xrt:: first class objects, e.g. xrt::hw_context.   From a run-time point of view, xrt_coreutil loads xrt_core, so at run time xrt_core is lower than xrt_coreutil, but coding wise it is reversed.  I like xrt_core using the handle classes only, but I have found this very difficult to do  without duplicating much code to each shim.  For example for the xclbin , yes, a hw context is associated with an xclbin, but where should this xclbin be managed?  In every shim?  Or in shared xrt_coreutil?  I prefer the latter.

Reply:
Hi @stsoe , I too have same point, xrt_core should just use the handles. Can you please point me the code where this PR using the xrt:: first class objects in xrt_core.  I see one include "xrt/xrt_hw_context.h" which needs to be fixed.
All other places are using handles only.

Reply:
@chvamshi-xilinx My point was opposite.  xrt_core can and does use xrt:: first class objects.  I am saying that in this PR you are forcing xclbin  management upon every shim, which is not cool.

One could argue that the shim handle classes should contain the shared data, and maybe that would be better, but that's now how it is designed today; e.g. all shim handles are pure interfaces with no data members.

Reply:
Thanks @stsoe  for the offline discussion.
Posting  the conclusion from our discussion here.
1) remove the xclbin_uuid function from base hwctx class. 
2) Store uuid/xclbin in shim level hwctx class if needed. 
3) Update all the shim level APIs to use shim level handles.